### PR TITLE
Crossplatform enhancements

### PR DESCRIPTION
**Pull request by winterheart**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [x] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [x] Network changes
  - [ ] Other changes

### Description

Added more crossplatform features to core code - static method `D3::ChronoTimer::SleepMS()` for sleeping with desired delay.
Reworked module submodule to strip some ddio leftovers.

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

**Review by Lgt2x — APPROVED** [review]
code is (almost) good, no regression found on Linux

File: module/module.cpp [hunk 1]
```diff
@@ -91,245 +91,92 @@
...
-#else
-  #error Unsupported platform!
-#endif
+
+std::filesystem::path mod_GetRealModuleName(const std::filesystem::path &mod_filename) {
+  std::filesystem::path filename = mod_filename;
+  std::filesystem::path ext = mod_filename.extension();
+
+  if (ext.empty() || (stricmp(ext.u8string().c_str(), ".so") == 0) || (stricmp(ext.u8string().c_str(), ".dll") == 0) ||
+      (stricmp(ext.u8string().c_str(), ".msl") == 0) || (stricmp(ext.u8string().c_str(), ".dylib") == 0)) {
```
[the remark this answers is not in the document]
Comment (winterheart) [reply module/module.cpp]: Additionally reworked this part.

**Review by Lgt2x — APPROVED** [review]

**Comment by Lgt2x** [comment]
good to merge?

**Comment by winterheart** [comment]
Yes, resolved last merge conflict.

**Pull request merged**